resource/template/resource_test.go

```diff
@@ -40,6 +43,27 @@ keys = [
 ]
 `
 
+// getRandomGroup randomly selects a gid from the Groups the current user is in
+func getRandomGroup() (gid int, err error) {
+	ogid := os.Getgid()
+	gid = ogid
+
+	// pick a random group and set the file to that group
+	groups, err := os.Getgroups()
+	if err != nil {
+		return
+	}
+
+	// It is possible this never terminates assert the length is greater than 1
```

Comment:
This logic is a bit concerning. I think it'd be better to use a concrete group ID in the tests so we don't have to do funky branch logic like this.